sdk/appconfiguration/azure-spring-cloud-feature-management/README.md

```diff
@@ -259,6 +259,104 @@ Options are available to customize how targeting evaluation is performed across
+              - "Amanda"
+            groups:
+              -
+                 name: "Ring0"
+                 rolloutPercentage: 10
+              -
+                 name: "Ring1"
+                 rolloutPercentage: 2W0
+            defaultRolloutPercentage: 0
+feature-variant:
+  DiscountBanner:
+    Big:
+      size: 400
+      color: "#BBB"
+    Small:
+      size: 150
+      color: "#DDD"
+```
+
+A Dynamic Feature has two main parts the assigner and a list of variants. The assigner in this case `feature-management.DiscountBanner.assigner` is pointing the `FeatureFilter` `Microsoft.Targeting` that will evaluate the list of variants and choose which configuration-reference will be used. Each variant has; a name, configuration-reference, and assignment-parameters, and one of them needs to be set as the default variant. The name field is just meta data to reference the variant and just needs to be unique. The configuration-reference is a link to the configurations defined under `feature-variant` in the configuration file. **Note:** the colon notation is required when denoting parts of the configuration. If it is needed multiple colons can be used to reference sub configurations. i.e. `DiscountBanner:Big:size` would return the `Integer` 400.
+
+The assignment-parameters work similarly to the other `FeatureFilters`. It can take in a `Map` which contains the evaluation criteria. In the case of `Microsoft.Targeting` it contains the `Audience` object. This enables custom assigners to be created.
+
+The `Microsoft.Targeting` assigner uses three evaluation criteria to assign a variant; users, groups, and defaultRolloutPercentage. Assignment requires the definition of the `ITargetingContextAccessor` where a user defines how the user and group parameters are defined.
+
+If a users is listed in a variant then they are automatically given that variant. If they are in more than one variant they are returned to the first listed variant.
+
+Groups have a rollout percentage. Using the user and group information users in groups are bucketed into parts of the rollout. This enables the a rollout where 10 of a group gets a variant, then when it increases to 50% that 10% still gets the variant with an additional 40% of the group now getting the variant. The rolloutPercentage for a group is shared across all variants that use that group.
+
+If Variant A has Group "Dev" which is set to 50, and Variant B also has Group "Dev" which is also set to 50 then, if you need to increase Variant A to have 60% for "Dev" then Variant B needs to be decreased to 40 as the max rollout percentage is 100.
+
+In a different setup Variant A has "Dev" set to 50 and Variant B has it set to 40. Then Variant A is increased to 60, then 10% of Variant B will move to use Variant A while the last 10% will be added to Variant B. This 10% of users could have had either variant before depending on the defaultRolloutPercentage configured and the default variant if one isn't assigned.
```

Comment:
This depends on ordering of the variants. I think we need an actual example with configuration to explain this concept.